Pages/shop.js

```diff
@@ -0,0 +1,207 @@
+    }
+
+    get dressShopFilter(){
+        return browser.driver.findElement(by.css("#shop-filter-sub-list-inside :first-child"))
+    }
+
+    get labelPricesFromFilterByPrice(){
+        return browser.driver.findElement(by.css("#shop-filters-price-prices label"));
+    }
+
+    get selectedFilterPopUp(){
+        return browser.driver.findElement(by.className("active-name"))
+    }
+
+    
+
+    // ACTIONS
+
+    getTextFromSelectedFilterPopUp(){
+        console.log("This method gets text from selected filter pop up");
+        return this.selectedFilterPopUp.getText();
+    }
+    validateElements(){
+        return this.getTextFromSelectedFilterPopUp()
+            .then((filterText) => this.validateTextFromSelectedFilterPopUp(filterText))
+    }
+    validateTextFromSelectedFilterPopUp(filterText){
+        console.log("This method validates text from selected filter pop up");
+        if(filterText === data.titleJacket){
+            return expect(filterText).toBe(data.titleJacket)
+        }else if(filterText === data.colorToSearch){
+            return expect(filterText).toBe(data.colorToSearch)
+        }   
+    }
+
+    clickOnDressShopFilter(){
+        console.log("This method clicks on Dress filter from Product categories filter");
+        return this.dressShopFilter.click();
+    }
+
+    clickOnJacketShopFilter(){
+        console.log("This method clicks on jacket filter in Shop");
+        return this.jacketShopFilter.click();
+    }
+
+    getPricesFromFilterByPrice(){
+        console.log("This method gets prices bellow slider in Filter by Price");
+        return this.labelPricesFromFilterByPrice.getText();
+    }
+
+    validatePricesFromFilterByPrice(prices){
+        console.log("This method validates prices from Filter by Price");
+        if(prices.split(" ")[0] === data.twoHundredDolars && prices.split(" ")[2] === data.nineHundredDolars){
+            return expect(true).toBe(true);
+        } else{
+            return expect(true).toBe(false);
+        }
+    }
+
+    clickOnBidButton(){
+        console.log("This method clicks on bid button");
+        return this.bidButton.click();
+    }
+
+    clickOnCategoryFilter(number){
+        console.log("This method clicks on Fashion Filter");
+        return this.filterNumber(number).click();
+    }
+
+    clickOnLeftSlider(){
+        console.log("This method clicks on left slider");
+        return this.leftSliderButton.click();
+    }
+
+    clickOnRightSlider(){
+        console.log("This method clicks on right slider");
+        return this.rightSliderButton.click();
+    }
+
+    moveLeftSlider(number){
+        console.log("This method moves left price slider to the right");
+        for(let i=1;i<number;i++){
+           this.leftSliderButton.sendKeys(protractor.Key.ARROW_RIGHT)
+        }
+        return this.leftSliderButton.sendKeys(protractor.Key.ARROW_RIGHT)
+    }
+
+    moverightSlider(number){
```

Comment:
Can you try to create one method  "moveSlider(number, direction)" instead having two

Reply:
I create clickAndMoveSlider() method.